widget/toolbar.go

```diff
@@ -22,8 +22,16 @@ type ToolbarAction struct {
 
 // ToolbarObject gets a button to render this ToolbarAction
 func (t *ToolbarAction) ToolbarObject() fyne.CanvasObject {
-	t.button.Icon = t.Icon
-	t.button.OnTapped = t.OnActivated
+	if t.button == nil {
+		// lazy initialization of the button, when the ToolbarAction 
+		// has not been created via the constructor
+		t.button = createToolbarActionButton(t.Icon, t.OnActivated)
+	} else {
+		// synchronize properties
+		t.button.Icon = t.Icon
+		t.button.OnTapped = t.OnActivated
+	}
```

Comment:
I was unsure about this, as it violates the happy-path-left pattern. But I also did not want to duplicate the `return t.button` line. Let me know what you think/prefer.

Reply:
I think there is a case that would still generate a problem though. If I would keep it like it is now, and create the `ToolbarAction` without the constructor, and then would use any of the setters, there would still be a crash. So I would need to pull the lazy initialization in all of the setters as well, or synchronize the state once in the `ToolbarObject` function, after lazy initializing it there. But this would require a property for the Disabled state again in the `ToolbarAction` so I can also synchronize it after the lazy initialization in the`ToolbarObject` function....

Sigh... am I missing something obvious/simple here?

Reply:
The code already has this problem I realised - the constructor is optional and there is no guarantee that `ToolbarObject` is called before you call `Disable`.

It's possible that the easiest fix here is to embed `Button` instead of `*Button` so you know it is always initialised. Then it's just a case of ensuring the state of the embedded button is correct?

Reply:
My comment above likely means that you can avoid the nil-check here.